src/astro/files/locations/s3.py

```diff
@@ -0,0 +1,44 @@
+import os
+from typing import Dict, List, Tuple
+from urllib.parse import urlparse, urlunparse
+
+from astro.constants import FileLocation
+from astro.files.locations.base import Location
+from astro.utils.dependencies import BotoSession, s3
+
+
+class S3(Location):
```

Comment:
It's really nice how easy this will make adding future file locations as needed (especially if community members want to add a more esoteric one)